internal/pkg/source/source.go

```diff
@@ -75,12 +75,16 @@ func New(in ssa.Node, config classifier) *Source {
 // While traversing the graph we also look for potential sanitizers of this Source.
 // If the Source passes through a sanitizer, dfs does not continue through that Node.
 func (s *Source) dfs(n ssa.Node) {
-	s.preOrder = append(s.preOrder, n)
-	s.marked[n.(ssa.Node)] = true
+	instr, ok := n.(ssa.Instruction)
+	if ok && !s.reachableFromSource(instr) {
+		return
+	}
 
-	if instr, ok := n.(ssa.Instruction); ok {
+	if ok {
```

Comment:
Nit: Does this read better?

```golang
	if instr, ok := n.(ssa.Instruction); ok {
		if !s.reachableFromSource(instr) {
			return
		}
		s.record(instr)
	}
```

Maybe, maybe not.

Reply:
I think it does, thanks! :)